BothamNetworking/BothamAPIClient.swift

```diff
@@ -115,6 +115,15 @@ public class BothamAPIClient {
         return interceptedResponse
     }
 
+    private func mapNSErrorToBothamAPIClientError(error: NSError) -> BothamAPIClientError {
```

Comment:
To do this, the best approach is to change HTTPClient interface to return ``Result<HTTPResponse, BothamAPIClientError>``. Is this your suggestion?